velox/expression/SimpleFunctionAdapter.h

```diff
@@ -141,7 +141,37 @@ class SimpleFunctionAdapter : public VectorFunction {
       const TypePtr& outputType,
       EvalCtx* context,
       VectorPtr* result) const override {
-    ApplyContext applyContext{&rows, outputType, context, result};
+    auto* reusableResult = result;
+    // If result is null or not uniquely-referenced, check if one of the
```

Comment:
Why is the requirement that the result must be `not uniquely referenced`?

Reply:
If 'result' is multiply-references, then it cannot be reused. In this case, BaseVector::ensureWritable will allocate a new vector and copy positions not in 'rows' into it.

Reply:
> If 'result' is multiply-references, then it cannot be reused

This is what I would assume as well. So we should check for `uniquely-referenced` `result` then?
The condition should be `if (!reusableResult->get() || reusableResult->unique()) {`

Reply:
I saw that the test you added has NULL result. So the second case is not being tested.

Reply:
The logic here is to check if 'result' is reusable and in case it is not check if an argument can be re-used instead. Hence, I believe the check is correct: `if (!reusableResult->get() || !reusableResult->unique())`.

Reply:
Ah! Thanks for the clarification. Makes sense now.

Reply:
@majetideepak This question made me realize that if 'result' is not null, reusing arg for result is a bit trickier. It is possible that 'rows' doesn't cover all positions, in which case we need to preserve positions outside of 'rows'. This would happen in case of a conditional expression using if or switch. While doable, this would make the change more complex and require more testing, hence, I'm going to simply change the check to `if (!reusableResult->get())` for now.